### PR TITLE
Cleanup bnxt driver and add supported commands

### PR DESCRIPTION
**Pull request by shitalt**

### CODE REVIEW

File: core/drivers/bnxt/bnxt_images.c [hunk 1]
```diff
@@ -20,53 +20,101 @@
...
 
-	bnxt_info->bnxt_bspd_cfg_vaddr =
-			(uintptr_t)((vaddr_t)phys_to_virt(QSPI_BSPD_ADDR,
-							  MEM_AREA_IO_NSEC));
+	img_header = (struct bnxt_img_header *)src;
+	if (dst) {
+		dst_header = (struct bnxt_img_header *)dst;
+		memcpy((void *)dst, (void *)img_header,
+		       sizeof(struct bnxt_img_header));
+		dst += sizeof(struct bnxt_img_header);
```
[the remark this answers is not in the document]
Comment (shitalt) [reply core/drivers/bnxt/bnxt_images.c]: Done

File: core/drivers/bnxt/bnxt_fw.c [hunk 2]
```diff
@@ -673,8 +675,22 @@ TEE_Result bnxt_load_fw(int chip_type)
...
+
+	if (offset + len > BNXT_CRASH_LEN)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	s = (vaddr_t)phys_to_virt(BNXT_CRASH_SEC_MEM, MEM_AREA_RAM_SEC);
+	s += offset;
+
+	cache_op_inner(DCACHE_AREA_INVALIDATE, (void *)s, len);
+
+	memcpy((void *)d, (void *)s, len);
```
[the remark this answers is not in the document]
Comment (shitalt) [reply core/drivers/bnxt/bnxt_fw.c]: Done

**Comment by shitalt** [comment]
For "drivers: bnxt: Remove NS3Z specific code", will update the commit description as below

`drivers: bnxt: Remove NS3Z specific code`
`Remove support for discontinued platform(NS3Z)`

**Comment by shitalt** [comment]
Applied Tags and force pushed

File: core/pta/bcm/bnxt.c [hunk 3]
```diff
@@ -17,6 +17,9 @@
 
 enum pta_bnxt_cmd {
 	PTA_BNXT_FASTBOOT = 0,
+	PTA_BNXT_HEALTH_STATUS,
+	PTA_BNXT_HANDSHAKE_STATUS,
+	PTA_BNXT_CRASH_DUMP_COPY,
```
Comment (jforissier) [inline core/pta/bcm/bnxt.c]: Minor: would be good to document the parameters for these new commands, it's good practice, but not strictly required either.

File: core/pta/bcm/bnxt.c [hunk 4]
```diff
@@ -45,6 +48,58 @@ static void close_session(void *session_context __unused)
...
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	d = (uint32_t *)params[0].memref.buffer;
+
+	if (!d)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	bnxt_copy_crash_dump((uint8_t *)d, params[1].value.a,
+			     params[1].value.b);
```
Comment (jforissier) [inline core/pta/bcm/bnxt.c]: Wait! If I understand correctly `a` and `b` are an offset and a size, but nowhere you check that the memory is accessible to the caller.

Comment (shitalt) [reply core/pta/bcm/bnxt.c]: Yes, The checks are implemented inside `bnxt_copy_crash_dump` API. 
I need to take care of the return value.

Comment (jforissier) [reply core/pta/bcm/bnxt.c]: `d` is not checked in relation with offset and size.

**Comment by shitalt** [comment]
Addressed Reivew comments
Rebased and applied tags

File: core/pta/bcm/bnxt.c (hunk 3 above)
Comment (shitalt) [reply core/pta/bcm/bnxt.c]: Done

File: core/pta/bcm/bnxt.c (hunk 4 above)
Comment (shitalt) [reply core/pta/bcm/bnxt.c]: Added check for memref.size against requested size

Comment (jforissier) [reply core/pta/bcm/bnxt.c]: OK LGTM

**Pull request merged**